db/sqlinterfaces.c

```diff
@@ -708,6 +708,49 @@ void sqlinit(void)
         abort();
 }
 
+static char *vtable_lockname(const char *table)
```

Comment:
I'm thinking that this part could be integrated into the system table creation routine. The system table lock type can be an additional argument of the creation routine. And then in this function, we could just look up the `table` in system table hashtable (just like how we do it in `select * from comdb2_systabs`), and return its lock type. This way it'd be easier to maintain.

Reply:
wow great idea .. actually ..